File: NumberGame.java
import java.util.Random;
import java.util.Scanner;

public class NumberGame {
    private static final int MAX_ATTEMPTS = 10;
    private static final int MAX_NUMBER = 100;
    private static final int MIN_NUMBER = 1;

    public static Integer playRound(Scanner scanner) {
        Random random = new Random();
        int number = random.nextInt(MAX_NUMBER - MIN_NUMBER + 1) + MIN_NUMBER;
        int attempts = 0;

        System.out.println("I've generated a random number between " + MIN_NUMBER + " and " + MAX_NUMBER + ". Can you guess it?");
        System.out.println("You have " + MAX_ATTEMPTS + " attempts.");

        while (attempts < MAX_ATTEMPTS) {
            System.out.print("Enter your guess: ");
            if (!scanner.hasNextLine()) {
                System.out.println("No more input. Ending round.");
                break;
            }
            try {
                int guess = Integer.parseInt(scanner.nextLine());
                attempts++;

                if (guess == number) {
                    System.out.println("Congratulations! You guessed it in " + attempts + " attempts.");
                    return attempts;
                } else if (guess < number) {
                    System.out.println("Too low!");
                } else {
                    System.out.println("Too high!");
                }
            } catch (NumberFormatException e) {
                System.out.println("Please enter a valid integer.");
            }
        }

        if (attempts >= MAX_ATTEMPTS) {
            System.out.println("Sorry, you've used all " + MAX_ATTEMPTS + " attempts. The number was " + number + ".");
        }
        return attempts > 0 && attempts <= MAX_ATTEMPTS ? attempts : null;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int totalScore = 0;
        int roundsPlayed = 0;

        while (true) {
            System.out.println("\n--- New Round ---");
            Integer attempts = playRound(scanner);
            if (attempts != null) {
                // Score: higher for fewer attempts, max 100 points
                int score = Math.max(0, 100 - (attempts - 1) * 10);
                totalScore += score;
                roundsPlayed++;
                System.out.println("Round score: " + score);
            } else {
                System.out.println("No score for this round.");
            }

            System.out.print("Do you want to play another round? (y/n): ");
            if (!scanner.hasNextLine()) {
                break;
            }
            String playAgain = scanner.nextLine().toLowerCase();
            if (!playAgain.equals("y")) {
                break;
            }
        }

        System.out.println("\nGame over! You played " + roundsPlayed + " rounds with a total score of " + totalScore + ".");
        scanner.close();
    }
}

File: number_game.py
import random

def play_round(max_attempts=10):
    """
    Play one round of the number guessing game.
    Returns the number of attempts used if guessed correctly, else None.
    """
    number = random.randint(1, 100)
    attempts = 0
    print("I've generated a random number between 1 and 100. Can you guess it?")
    print(f"You have {max_attempts} attempts.")

    while attempts < max_attempts:
        try:
            guess = int(input("Enter your guess: "))
            attempts += 1

            if guess == number:
                print(f"Congratulations! You guessed it in {attempts} attempts.")
                return attempts
            elif guess < number:
                print("Too low!")
            else:
                print("Too high!")
        except ValueError:
            print("Please enter a valid integer.")

    print(f"Sorry, you've used all {max_attempts} attempts. The number was {number}.")
    return None

def main():
    total_score = 0
    rounds_played = 0

    while True:
        print("\n--- New Round ---")
        attempts = play_round()
        if attempts is not None:
            # Score: higher for fewer attempts, max 100 points
            score = max(0, 100 - (attempts - 1) * 10)
            total_score += score
            rounds_played += 1
            print(f"Round score: {score}")
        else:
            print("No score for this round.")

        play_again = input("Do you want to play another round? (y/n): ").lower()
        if play_again != 'y':
            break

    print(f"\nGame over! You played {rounds_played} rounds with a total score of {total_score}.")

if __name__ == "__main__":
    main()
